index.js

```diff
@@ -1,23 +1,36 @@
-'use strict';
-
-var glob = require('glob');
-var gulp = require('gulp');
+var argv = require('minimist')(process.argv.slice(2));
+var fs = require('fs');
 var path = require('path');
 
-module.exports = function (pattern, options) {
-    options = options || {};
-    options = {
-        base: path.resolve(options.base || './')
-    };
+module.exports = function (options) {
+  options = options || {};
+  var bases = options.base || './';
+  var gulp = options.gulp || require('gulp');
+  var tasks = argv._;
+
+  if (typeof bases === 'string') {
+    bases = [bases];
+  }
 
-    glob.sync(path.join(options.base, pattern)).forEach(function (file) {
-        var task = require(file);
+  tasks.forEach(function (task) {
+    bases.forEach(function (base) {
+      var baseTask = path.join(process.cwd(), base, task);
+      var baseTaskJs = baseTask + '.js';
+      if (fs.existsSync(baseTaskJs)) {
+        var taskFunc = require(baseTask);
+
+        if (taskFunc.private) {
+          return;
+        }
 
-        if (task.autotask === false) {
-            return;
+        if (!taskFunc.dependencies) {
+          taskFunc.dependencies = [];
         }
 
-        var name = file.replace(options.base + '/', '').replace('.js', '');
-        gulp.task(name, task.deps || task, task.deps ? task : undefined);
+        gulp.task(task, taskFunc.dependencies, function () {
+          return taskFunc(argv);
```

Comment:
Command line args are now passed in to the task function as an options argument.